crates/ark/tests/data_explorer.rs

```diff
@@ -150,6 +157,543 @@ fn socket_rpc(
     socket_rpc_request::<DataExplorerBackendRequest, DataExplorerBackendReply>(&socket, req)
 }
 
+/// Test setup helper that reduces boilerplate for common test initialization
+struct TestSetup {
+    socket: socket::comm::CommSocket,
+}
+
+impl TestSetup {
+    fn new(dataset: &str) -> Self {
+        Self {
+            socket: open_data_explorer(String::from(dataset)),
+        }
+    }
+
+    fn from_expression(expr: &str, bind: Option<&str>) -> anyhow::Result<Self> {
+        Ok(Self {
+            socket: open_data_explorer_from_expression(expr, bind)?,
+        })
+    }
+
+    fn socket(&self) -> &socket::comm::CommSocket {
+        &self.socket
+    }
+}
+
+/// Request builder helpers to reduce repetitive request construction
+struct RequestBuilder;
```

Comment:
Not a big deal but in Rust a `Builder` struct generally denotes a multi-step ctor where you call multiple methods to specify your object, ending with `finish()` or `create()`.

Another way of setting this up would have been a `request` submodule that isn't imported, then you call it like `request::get_schema()`.

Just a note, no need to change now.